test/Microsoft.Azure.WebJobs.Host.UnitTests/FakeQueue/FakeQueueClient.cs

```diff
@@ -30,19 +31,67 @@ public Task FlushAsync(CancellationToken cancellationToken = default(Cancellatio
 
         void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
         {
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
             IConverterManager cm = context.Config.GetService<IConverterManager>();
             cm.AddConverter<string, FakeQueueData>(x => new FakeQueueData { Message = x });
             cm.AddConverter<FakeQueueData, string>(msg => msg.Message);
             cm.AddConverter<OtherFakeQueueData, FakeQueueData>(OtherFakeQueueData.ToEvent);
 
             IExtensionRegistry extensions = context.Config.GetService<IExtensionRegistry>();
 
-            var bindingProvider = new FakeQueueBindingProvider(this, cm);
-            extensions.RegisterExtension<IBindingProvider>(bindingProvider);
+            var bf = new BindingFactory2(nameResolver, cm);
```

Comment:
This part is the most interesting: this is the what the extension config would write.